Upgraded EUI to v76.0.0

JasonStoltz opened the pull request:
👋 Hi all - the biggest breaking change of this PR is around two icon type changes/renames.

1. ⚠️ **The  `alert` icon is now named `warning`**
    - <img width="103" alt="" src="https://user-images.githubusercontent.com/549407/223561599-8913e88c-676f-47cd-aaed-81b64783bd81.png" align="middle">
    - This change should have been automatically converted on your behalf by the EUI team, **but if for some reason** we missed making this conversion in this PR and your icon(s) are now broken, please ping us or let us know in this PR (or fix yourself after this PR merges).
    - In some cases, teams were using this icon for error messages, alongside the `danger` color. In those cases, we opinionatedly changed those icon usages to the new `error` icon instead of using the old alert/warning icon.

2. 🛑 **The `crossInACircleFilled` icon has been removed, and a new `error` icon added**
    - <img width="84" alt="" src="https://user-images.githubusercontent.com/549407/223561892-4406bdf6-1a55-49ac-85ad-3a11eb7c090d.png" align="middle">
    - The conversion for this breaking change was not straightforward. This was the path we used to determine what to change `crossInACircleFilled` usages to:
       - If the icon was associated with errors or error messages, we changed it to the new `error` icon.
       - If a "delete" action was associated with this icon, we changed it to the `trash` icon instead.
       - If a "clear" action was associated with this icon, we changed it to just the `cross` icon, or in some cases `minusInCircleFilled` (if used alongside `plusInCircleFilled`).
    - Again, if we made a mistake during this conversion or missed your plugin, please feel free to ping us.

## Summary

`eui@75.1.2` ⏩ `eui@76.0.2`

## [`76.0.2`](https://github.com/elastic/eui/tree/v76.0.2)

**Bug fixes**

- Added a legacy `alert` alias for the `warning` `EuiIcon` type ([#6640](https://github.com/elastic/eui/pull/6640))

## [`76.0.1`](https://github.com/elastic/eui/tree/v76.0.1)

**Bug fixes**

- Fixed broken icons on all `isInvalid` form controls ([#6629](https://github.com/elastic/eui/pull/6629))

## [`76.0.0`](https://github.com/elastic/eui/tree/v76.0.0)

- Added `pivot` glyph to `EuiIcon` ([#6605](https://github.com/elastic/eui/pull/6605))
- Added the `displayHeaderCellProps` API to `EuiDataGrid`'s columns, which allows passing custom props directly to column header cells ([#6609](https://github.com/elastic/eui/pull/6609))
- Added the new `headerCellProps`/`footerCellProps` APIs to `EuiDataGrid`'s control columns, which allows passing custom props directly to control column header or footer cells ([#6609](https://github.com/elastic/eui/pull/6609))
- Added a new `footerCellRender` API to `EuiDataGrid`'s control columns, which allows completely customizing control column rendering (previously rendered an empty cell) ([#6609](https://github.com/elastic/eui/pull/6609))
- Updated the styling of nested ordered lists in `EuiText` to align with GitHub's list style, which is a popular format used in Markdown or MDX formatting ([#6615](https://github.com/elastic/eui/pull/6615))
- Added a margin-bottom property exclusively to the direct child `ul` and `ol` elements of the `EuiText` component ([#6615](https://github.com/elastic/eui/pull/6615))
- Fix issue with badges appearing within an `EuiBadgeGroup`, where the CSS rule to override the `margin-inline-start` was not being applied correctly due to the order of appearance in the CSS rules ([#6618](https://github.com/elastic/eui/pull/6618))

**Bug fixes**

- Fixed `EuiDataGrid` footer control columns rendering with cell expansion popovers when they should not have been ([#6609](https://github.com/elastic/eui/pull/6609))
- Fixed an `EuiSkipLink` bug where main content loading in progressively/dynamically after the skip link rendered was not being correctly focused ([#6613](https://github.com/elastic/eui/pull/6613))

**Breaking changes**

- Renamed `EuiIcon`'s `alert` to `warning` ([#6608](https://github.com/elastic/eui/pull/6608))
- Removed `EuiIcon`'s `crossInACircleFilled` in favor of `error` ([#6608](https://github.com/elastic/eui/pull/6608))

JasonStoltz committed:
Upgraded EUI to v76.0.0

JasonStoltz committed:
Updated snapshots

JasonStoltz committed:
Patched to 76.0.1

elasticmachine commented:
Pinging @elastic/eui-design (EUI)

JasonStoltz committed:
Merge branch 'main' into eui-v76.0.0

azasypkin commented:
> Breaking changes
> Renamed EuiIcon's alert to warning (https://github.com/elastic/eui/pull/6608)

@JasonStoltz won't it break [many places in Kibana](https://github.com/search?q=repo%3Aelastic%2Fkibana%20%27alert%27&type=code) or `alert` is still considered a valid icon type for backward-compatibility?

cee-chen commented:
> @JasonStoltz won't it break [many places in Kibana](https://github.com/search?q=repo%3Aelastic%2Fkibana%20%27alert%27&type=code) or alert is still considered a valid icon type for backward-compatibility?

@azasypkin is correct, we need to update all instances of `type="alert"` or `iconType="alert"` (or variable usages) to `warning` as part of this PR. @miukimiu or @daveyholler would either of you be able to help with that?

daveyholler commented:
@cee-chen I'll take that.

JasonStoltz commented:
Sorry ya'll, totally missed that.

JasonStoltz committed:
Fixed snapshot

JasonStoltz committed:
Merge branch 'main' into eui-v76.0.0

daveyholler committed:
alert → warning, crossInACircleFilled → error

[CI] Auto-commit changed files from 'node scripts/eslint --no-cache --fix'

cee-chen committed:
Convert more instances of `crossInACircleFilled`

- Errors -> `error`
- Delete actions -> `trash`
- Clear actions -> `cross` or `minusInCircleFilled` (if used alongside `plusInCircleFilled`)

cee-chen committed:
Convert instances of `alert` usages combined w/ error vars & danger color to use new `error` icon instead

cee-chen committed:
Convert remaining instances of `alert` icon types to `warning`

cee-chen committed:
Update tests, mocks, and snapshots with `alert`->`warning`

elasticmachine commented:
Pinging @elastic/apm-ui (Team:APM)

elasticmachine commented:
Pinging @elastic/fleet (Team:Fleet)

cee-chen committed:
Merge branch 'main' into eui-v76.0.0

cee-chen committed:
Fix more snapshots

cee-chen commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into eui-v76.0.0

gergoabraham reviewed (approved):
defend workflows related changes look good 👍

walterra reviewed (approved):
ML related changes LGTM

benakansara reviewed (approved):
AO changes LGTM.
@elastic/actionable-observability We might want to change the icon type to `error` when showing error message. But I believe we can do that in separate PR if needed.

nreese reviewed (approved):
@elastic/kibana-presentation  changes LGTM
code review only

kc13greiner commented on x-pack/plugins/security/server/prompt_page.tsx:
```diff
@@ -65,7 +65,7 @@ export function PromptPage({
           <EuiPageBody>
             <EuiPageContent verticalPosition="center" horizontalPosition="center">
               <EuiEmptyPrompt
-                iconType="alert"
```
<img width="803" alt="Screenshot 2023-03-08 at 10 14 05 AM" src="https://user-images.githubusercontent.com/21210601/223752429-5d49ea6a-9583-499b-a668-6cb68c56a6b4.png">

The icon does not appear properly on this page.

cee-chen replied:
Should be fixed now in 1351cba - thanks a million for catching this @kc13greiner!

kc13greiner reviewed (changes requested):
There seems to be an issue on the prompt page; it is rendered server side, so maybe that has something to do with it?

yuliacech reviewed (approved):
Management team changes LGTM, only renaming `alert` to `warning`, didn't test locally

opauloh reviewed (approved):
csp / awp Icons LGTM

daveyholler commented on x-pack/plugins/synthetics/public/apps/synthetics/components/monitor_details/monitor_errors/errors_icon.tsx:
```diff
@@ -11,5 +11,5 @@ import { useMonitorErrors } from '../hooks/use_monitor_errors';
   const { hasActiveError } = useMonitorErrors();
 
-  return hasActiveError ? <EuiIcon type="alert" color="danger" /> : null;
+  return hasActiveError ? <EuiIcon type="error" color="danger" /> : null;
```
Hey @justinkambic! Its a great question. Intent is important here and its ultimately up to your team to decide which icon makes the most sense in your application's flow. The (yet to be documented) guidance from a design system perspective is as follows:

- The `error` icon should be used to indicate blocking issues that prevent a user from moving forward in a workflow
- The `warning` icon should be used to caution users against something that _should_, but does not necessarily _need_, to be addressed before moving forward.

I hope that helps clarify our direction. Our suggestion in the lines above were inferred from the surrounding code. We're happy to change this implementation if `warning` makes more sense in your case.

cee-chen replied:
https://github.com/elastic/kibana/pull/152506/commits/9b2eb8f4c37e32063ce637c1382ee141ae44e2c2 I've switched this back to the `warning` (old alert) icon per your comment @justinkambic, but just want to reiterate what Davey said - the renames aren't just renames, they're also requests for teams and designers to reconsider your usage across the board and whether you're using the right icon for the right scenario. It might be worth revisiting (after this PR merges) with your designer.

dominiqueclarke replied:
This isn't actually an error in workflow, but an indication of degradation in the synthetic monitor itself and meant to warn users that they may need to review errors in the services they are monitoring.

cee-chen replied:
Gotcha, thanks for taking the time to confirm Dominique!

justinkambic replied:
++ thanks for the added context @dominiqueclarke ❤️

daveyholler reviewed (approved):
Lookin' good. I dig it.

cee-chen committed:
Merge remote-tracking branch 'upstream/main' into eui-v76.0.0

cee-chen committed:
[PR feedback] Fix server-side warning icon

cee-chen committed:
[PR feedback] Switch `error` icon to `warning`

cee-chen committed:
Actually fix SSR warning icon

- fix missing renames

cee-chen commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into eui-v76.0.0

kc13greiner reviewed (approved):
LGTM!

justinkambic reviewed (approved):
Synthetics changes LGTM!

cee-chen commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into eui-v76.0.0

cee-chen committed:
Update to latest EUI v76.0.2 backport

cee-chen committed:
Merge remote-tracking branch 'upstream/main' into eui-v76.0.0

cee-chen commented:
Pinging all remaining CODEOWNERS:

@elastic/sec-cloudnative-integrations @elastic/security-detections-response @elastic/security-solution-platform @elastic/security-detections-response @elastic/security-solution-platform @elastic/security-detections-response-alerts @elastic/security-detections-response-rules @elastic/security-threat-hunting-explore @elastic/security-threat-hunting-investigations @elastic/appex-sharedux @elastic/kibana-visualizations @elastic/response-ops @elastic/protections-experience @elastic/infra-monitoring-ui @elastic/fleet @elastic/apm-ui

We've determined any failing or flaky tests are unrelated to our EUI changes (since after rerunning / merging latest multiple times, the tests that fail change each time), and will be asking Kibana Ops to admin merge this PR by Friday EOD. If you could take a second to review your owned files before then, we would hugely appreciate it. If not, and you notice any issues (primarily around the breaking changes of the new `warning` and `error` icons), you should be able to fix them after this PR merges and well before the next feature freeze.

jbudz committed:
Merge branch 'main' into eui-v76.0.0

cauemarcondes reviewed (approved):
Profiling changes LGTM

nchaulet reviewed (approved):
Fleet changes LGTM

michaelolo24 reviewed (approved):
Investigations changes. LGTM!

drewdaemon reviewed (approved):
Visualizations changes look good.

I tested importing an 8.6 visualization with alert icons and they remained unchanged. The other icon substitutions make sense.

<img width="1560" alt="Screenshot 2023-03-10 at 2 35 26 PM" src="https://user-images.githubusercontent.com/315764/224424639-b4010ce4-da15-4778-9b40-946774e612c3.png">

jbudz committed:
Merge branch 'main' into eui-v76.0.0

cee-chen commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into eui-v76.0.0
